### ISSUE TITLE
GPU utilization

### ISSUE THREAD

**Issue by LiuYustar** [Kaiseem/SLAug#2]
Hello blogger, when I run this code to train, I find that the utilization of my GPU is extremely low. Is this normal, and if so, how to improve its utilization?
![image](https://user-images.githubusercontent.com/59524646/211878683-6f0159e2-17e4-4121-9cf7-cb810ee5bd58.png)

**Comment by Kaiseem** [comment]
Hi, it's not normal. It seems that the CPU computation is insufficient. I suggest to decrease the nTimes in https://github.com/Kaiseem/SLAug/blob/c9020793d55f3b610594616ce271089d06a5913e/dataloaders/location_scale_augmentation.py#L6 , e.g., nTimes=1000. It may degrade the performance slightly, but it can decrease the demand of CPU computation significantly.

**Comment by LiuYustar** [comment]
Hello, I just tried the method you said, but the utilization rate of the GPU is not high, only up to 20%. It should be noted that I have not changed any code at all, just downloaded the processed data. In addition, I also found that during training, the training time of each epoch needs 44s. I am not sure whether this time is normal, and I hope you can give some suggestions, thank you.

**Comment by LiuYustar** [comment]
To add, the command I run is **python main.py --base configs/efficientUnet_bSSFP_to_LEG.yaml --seed 23**
And I set the batch_size and num_workers of the configuration file to 4

**Comment by Kaiseem** [comment]
Hi, i strongly suggest you to check the usage of CPU, as my colleagues found it could be very slow when trained on the server with good GPU but poor CPU. Since we follow CSDG augmentation which is totally run on CPU, i am afraid that i can only suggest you to update the CPU.

Meanwhile, the training time per epoch is expected about 20 seconds on CT-MRI in my experiences. I don't remember the exact time, but i can train the model on CT-MRI two or three times a day on one 3090.

**Comment by LiuYustar** [comment]
Ok, thank you very much for your reply, I will seriously consider what you said

**Comment by fruit2rbj** [comment]
I found it is very slow when trained using numpy without mkl acceleration. The training time per epoch is about 26s with mkl, but it takes 4 minutes without mkl.
